### ISSUE TITLE
S3 endpointOverride needs to be in DataAddess 

### ISSUE THREAD

**Issue by reisman234** [eclipse-dataspaceconnector/DataSpaceConnector#1903]
# Feature Request
The current implementation to configure the `endpointOverride` parameter is inappropriate. If it is configured in the property file, the connector is only able to make transfer to that specified endpoint. 
With that, it is not possible to make a transfer from a private MINIO to a real AWS s3 location or the other way around. 

So IMHO this parameter needs to be an optional one which could be specified in a `DataAddress`. 

## Which Areas Would Be Affected?

- s3-core
- s3-data-plane

## Why Is the Feature Desired?
Improvement of the s3-data-plane

## Solution Proposal
- simple solution: 
  - add setters for endpointOverride in `AwsClientProviderConfiguration` and `AwsClientProvider`
  - add endpointOverride to `S3BucketSchema`
  - get Property(ENDPOINT_OVERRIDE) from DataAddress in S3Sink (`createSink()`) and S3Source(`createSource()`) and set that via setter.
  - **beware**: S3Sink and S3Source have the same ref to AwsClientProvider, so they will use the same `AwsClientProviderConfiguration` -> RACE-CONDITION! But at the moment, it looks for me `createSource()` and `createSink()` is called in sequence.

## Type of Issue
**improvement**
## Checklist

- [ ] assigned appropriate label?
- [x] **Do NOT select a milestone or an assignee!**

**Comment by ndr-brt** [comment]
I agree, that `endpointOverride` was mainly introduced for testing purposes, as you reported, that's not really useful in a "push" data transfer scenario (but it is for the "pull" one).
I didn't have the time to check your PR, but I'm available to help to get this solved.

**Comment by reisman234** [comment]
Thanks for the response!
So you will do the required changes and don't need any input from my side?
